Lesson-6/BankAccount.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+
+namespace Homework6
+{
+    public class BankAccount
+    {
+        public enum TypeAccount 
+        {
+            CURRENCY = 1,
+            BROKER,
+            SAVINGS,
+        }
+        private int numberAccount = numAcc;
+        private decimal balance;
+        private TypeAccount type;
+        static int numAcc = 1;
+
+
+        public BankAccount(int type,decimal balance)
+        {
+            CreateNumAccount();
+
+            switch (type)
+            {
+                case 1:
+                    this.type = TypeAccount.CURRENCY;
+                    break;
+                case 2:
+                    this.type = TypeAccount.BROKER;
+                    break;
+                case 3:
+                    this.type = TypeAccount.SAVINGS;
+                    break;
+                default:
+                    throw new Exception("Incorrect input value");
+            }
+
+
+            if (balance >= 0)
+            {
+                this.balance = balance;
+            }
+            else
+            {
+                throw new Exception("Incorrect input values");
+            }
+
+        }
+
+        public BankAccount(int type)
+        {
+            CreateNumAccount();
+
+            switch (type)
+            {
+                case 1:
+                    this.type =  TypeAccount.CURRENCY;
+                    break;
+                case 2:
+                    this.type = TypeAccount.BROKER;
+                    break;
+                case 3:
+                    this.type = TypeAccount.SAVINGS;
+                    break;
+                default:
+                    throw new Exception("Incorrect input value");
+            }
+        }
+
+        public BankAccount(decimal balance)
+        {
+            CreateNumAccount();
+
+            if (balance >= 0)
+            {
+                this.balance = balance;
+            }
+            else
+            {
+                throw new Exception("Incorrect input values");
+            }
+        }
+
+        public BankAccount()
+        {
+            CreateNumAccount();
+        }
+
+        public static bool operator ==(BankAccount b1, BankAccount b2)
+        {
+            if(b1 is TypeAccount && b2 is TypeAccount)
+                return b1.Types == b2.Types;
+
+            bool value = b1.Balance == b2.Balance ? true : false;
+            return value;
+
+        }
+        public static bool operator !=(BankAccount b1, BankAccount b2)
+        {
+            if(b1 is TypeAccount && b2 is TypeAccount)
+                return b1.Types != b2.Types;
+
+            bool value = b1.Balance != b2.Balance ? true : false;
```

Comment:
Этот индусский код зачем вообще?
```suggestion
            bool value = b1.Balance != b2.Balance;
```
результат оператора `==` - это и так bool-значение.